TypeCobol/Compiler/Diagnostics/TypeCobolChecker.cs

```diff
@@ -535,15 +535,17 @@ private static void Check(Node node, [NotNull] FunctionCall call, [NotNull] IPro
                         DiagnosticUtils.AddError(node, m);
                     }
 
-                    if (actualDataDefinition.IsSynchronized != expected.IsSynchronized)
+                    bool actualIsSynchronized = actualDataDefinition.Synchronized != null;
+                    bool expectedIsSynchronized = expected.Synchronized != null;
+                    if (actualIsSynchronized != expectedIsSynchronized)
                     {
                         var m =
                            string.Format(
                                "Function '{0}' expected parameter '{1}' {2} and received '{3}' {4}",
                                 call.FunctionName, expected.Name,
-                                expected.IsSynchronized ? "synchonized" : "not synchronized",
+                                expectedIsSynchronized ? "synchonized" : "not synchronized",
                                 callArgName ?? string.Format("position {0}", c + 1),
-                                actualDataDefinition.IsSynchronized ? "synchonized" : "not synchronized");
+                                actualIsSynchronized ? "synchonized" : "not synchronized");
                         DiagnosticUtils.AddError(node, m);
                     }
 
```

Comment:
As discussed, it's a good idea to add an extra information regarding sync flags in the diagnostics :

- it will deliver more acute information
- we can then add a test to validate the PR

Reply:
See b2a09b3eb0592aa24286448eea0180cc57d9bbbf.